### PR TITLE
Move `-install_name` linker flag from the `{ios,tvos}_framework` macro into the rule implementation.

### PR DESCRIPTION
**Pull request by thii**
PiperOrigin-RevId: 382086173
(cherry picked from commit 172ad76549d732b3380bef6a7c255c32cc4462d3)

Conflicts:
	apple/internal/ios_rules.bzl
	apple/internal/tvos_rules.bzl
	apple/ios.bzl
	apple/tvos.bzl

### CODE REVIEW

**Review by keith — APPROVED** [review]

**Comment by thii** [comment]
Depends on https://github.com/bazelbuild/rules_apple/pull/1188

File: apple/internal/ios_rules.bzl [hunk 1]
```diff
@@ -680,6 +665,25 @@ def _ios_framework_impl(ctx):
         res_attrs = ["resources"],
     )
 
+    extra_linkopts = [
+        "-dynamiclib",
+        "-Wl,-install_name,@rpath/{name}{extension}/{name}".format(
```
Comment (brentleyjones) [inline apple/internal/ios_rules.bzl:670]: I used this version to work around https://github.com/bazelbuild/bazel/blob/1c3a2456c95fd19974a5b2bd33c5ebdb2b2277e4/src/main/java/com/google/devtools/build/lib/rules/cpp/LinkCommandLine.java#L295 causing an issue. Think this is bug worthy? One nice thing about doing it this way: it's the same argument that was being sent before.

Comment (thii) [reply apple/internal/ios_rules.bzl:670]: Thanks. Yes I think we should report it.

**Comment by brentleyjones** [comment]
Hmm, it should have re-branched, not closed this...

**Comment by brentleyjones** [comment]
@thii I'm in the middle of a rebase right now. Care to open a new one with the changes that exist already in your repo?

**Comment by brentleyjones** [comment]
The last two commits just need to be rebased onto master now.

File: apple/internal/ios_rules.bzl (hunk 1 above)
Comment (brentleyjones) [reply apple/internal/ios_rules.bzl:670]: https://github.com/bazelbuild/bazel/issues/14316

**Pull request merged**